data.rs

```diff
@@ -45,14 +45,23 @@ pub struct PSP22Data {
 
 impl PSP22Data {
     /// Creates a token with `supply` balance, initially held by the `creator` account.
-    pub fn new(supply: u128, creator: AccountId) -> PSP22Data {
+    pub fn new(supply: u128, creator: AccountId) -> (PSP22Data, Vec<PSP22Event>) {
         let mut data = PSP22Data {
             total_supply: supply,
             balances: Default::default(),
             allowances: Default::default(),
         };
         data.balances.insert(creator, &supply);
-        data
+        let events = if supply > 0 {
```

Comment:
There is also another issue here that if `supply` is `0` the balance should not be inserted.

I think the cleanes here is in line `54` just do:
`let events = data.mint(creator, supply)?;`
and get rid of the rest.

Reply:
Also `total_supply` should be initialized at `0` in that case.

Reply:
Now I see that `data.mint(...)` needs an `unwrap()` but that's fine, since it's guaranteed to succeed.